cradlepoint/pages/4TestPlanDetails.jsx

```diff
@@ -173,3 +173,56 @@ export default function TestPlanDetails(props) {
         </>
     )
 }
+
+export async function getServerSideProps(context) {
+    /* 
+       Gets Data for Test Plan Details
+       TODO: Error Check await call
+       TODO: Refactor out fetch call
+    */
+    const res = await fetch(`${process.env.HOST}/api/getTestPlan?testPlanId=`+context.query.TestPlanId);
+    const testPlanData = await res.json().then((data) => {
+        return {...data[0], };
+    });
+
+    /* 
+       Gets Data for BOM Table
+       TODO: Error Check await call
+       TODO: Refactor out fetch call
+    */
+    var BOM_entries = [];
+    for(var i = 0; i < testPlanData.summaryBOM.length; i++) {
+        const deviceRes = await fetch(`${process.env.HOST}/api/getDevice?deviceId=`
+        +testPlanData.summaryBOM[i].deviceId);
+        const deviceData = await deviceRes.json();
+        const BOM_entry = { 
+            ...deviceData[0],
+            "isOptional": testPlanData.summaryBOM[i].isOptional,
+            "quantity": testPlanData.summaryBOM[i].quantity,
+        };
+        BOM_entries.push(BOM_entry);
+    }
```

Comment:
I have a way to grab all this data with test plans from the database that means we don't have to map anything, check out my branch

Reply:
Please refer to the fix so I don't have to find it in your code.

Reply:
getTestPlan endpoint has been modified so that it pulls the BOM data with it. Use this call: http://localhost:3000/api/getTestPlan?_id=61724e5599915be1b771acb2 and see the data in summaryBOM to notice changes